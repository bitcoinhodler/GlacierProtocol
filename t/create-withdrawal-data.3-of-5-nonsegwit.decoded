{
  "txid": "864559e8a57cf9fca04e121c5ccc4343998bada766637e26a0d572b7b2c31004",
  "hash": "864559e8a57cf9fca04e121c5ccc4343998bada766637e26a0d572b7b2c31004",
  "version": 2,
  "size": 479,
  "vsize": 479,
  "weight": 1916,
  "locktime": 0,
  "vin": [
    {
      "txid": "f18c44eaf7688eba8526c46a239f332ca0bbbc436212469f63ab4bc86b201dbb",
      "vout": 2,
      "scriptSig": {
        "asm": "0 304402207c642218b2e4399473f50f61832506f6072df02090a07a46b535ad3509088bc402207b41970dd63558473d85531fc70b8a79522719ff15a268ae39f7066502314686[ALL] 30440220483800ebe22d861bec585f2d6748aa267c57031657fadcc9cfba1e47beee166002203c4bf887d2f29605bc6d7adc0ce67376214cb3b507651a434cd17f9818e532fe[ALL] 304402200aafc5743aeef61ae3faca56f89fb6ca22cd7c8a7d0f0cba92665a29deefe25c022055a874f97ea0485f0e11e3a0a29b7800a8ff101eb6530cff9b3b086a3bc135fa[ALL] 53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae",
        "hex": "0047304402207c642218b2e4399473f50f61832506f6072df02090a07a46b535ad3509088bc402207b41970dd63558473d85531fc70b8a79522719ff15a268ae39f7066502314686014730440220483800ebe22d861bec585f2d6748aa267c57031657fadcc9cfba1e47beee166002203c4bf887d2f29605bc6d7adc0ce67376214cb3b507651a434cd17f9818e532fe0147304402200aafc5743aeef61ae3faca56f89fb6ca22cd7c8a7d0f0cba92665a29deefe25c022055a874f97ea0485f0e11e3a0a29b7800a8ff101eb6530cff9b3b086a3bc135fa014cad53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.19995210,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 b6c7733e23a271e57ca16e5cd885174917968db9 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99)#fxf0ktmw",
        "hex": "76a914b6c7733e23a271e57ca16e5cd885174917968db988ac",
        "address": "mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99",
        "type": "pubkeyhash"
      }
    }
  ]
}
