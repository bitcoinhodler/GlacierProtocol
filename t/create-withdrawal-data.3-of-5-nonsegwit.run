#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
2N75fd3tacsz3g25Eak6XnP7wizd8scVnGe
53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155ae
mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
1
0200000001cdd1ce67fd9655af40ab56a162c4045efee0b46019617006780b8b399e5432b2010000006a47304402201281cfeac9790be29e4472970a4bb35b0791bf23d38f74643fa95f118585f5760220508f5fde2779a4eabb6016e547a285b1a8dce42325fbb1760200f2270f1f55c10121037dc8c8636258079c0a55fe56e82d147ceff15aa450c8ff56c02fce13af70cebfffffffff05002d31010000000017a91460fee68955e188231d562dccec9e363501dac9578790409700000000001976a91446248cd28e43ce66475393b17dd2f54663fc09ab88ac002d31010000000017a91497c2ffdcdfc233a328751b46a47b781b1eec9b2d87002d31010000000017a914ced9b532efa26a61566852456a21c1e14986101c87002d31010000000017a9143f0adad853ec261b4fd7df2d3ef24fe5807dfeea8700000000
3
cQy39MvkMKeaEvG1wXejdtEvAdU381u9EBBmcKUJWbfn6QL82YvR
cUv81BVz1JWuA9uwECPvfsoyA1TqmFJBJaQUPYrxkdf44zQSiTZY
cVV1MBAgdAnrX8bjQWSJcgZAbVtprR8kgdfez858ejvXs3qZWqn2
10
y

y
INPUT
