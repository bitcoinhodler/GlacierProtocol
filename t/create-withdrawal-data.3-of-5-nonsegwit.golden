Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 BTC

How many private keys will you be signing this transaction with (at least 3 required to complete transaction)?
#: Key #1: Key #2: Key #3: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00005110 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00005110, you have 0.19994890 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.19995210 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 BTC in unspent inputs from cold storage address 2N75fd3tacsz3g25Eak6XnP7wizd8scVnGe
0.19995210 BTC going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00004790

Signing with private keys: 
cQy39MvkMKeaEvG1wXejdtEvAdU381u9EBBmcKUJWbfn6QL82YvR
cUv81BVz1JWuA9uwECPvfsoyA1TqmFJBJaQUPYrxkdf44zQSiTZY
cVV1MBAgdAnrX8bjQWSJcgZAbVtprR8kgdfez858ejvXs3qZWqn2


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
0200000001bb1d206bc84bab639f46126243bcbba02c339f236ac42685ba8e68f7ea448cf102000000fd88010047304402207c642218b2e4399473f50f61832506f6072df02090a07a46b535ad3509088bc402207b41970dd63558473d85531fc70b8a79522719ff15a268ae39f7066502314686014730440220483800ebe22d861bec585f2d6748aa267c57031657fadcc9cfba1e47beee166002203c4bf887d2f29605bc6d7adc0ce67376214cb3b507651a434cd17f9818e532fe0147304402200aafc5743aeef61ae3faca56f89fb6ca22cd7c8a7d0f0cba92665a29deefe25c022055a874f97ea0485f0e11e3a0a29b7800a8ff101eb6530cff9b3b086a3bc135fa014cad53210320849607eed265c09f51cda0e972426558276384870651250aa4e8638c334ced2102766c70472a8a3de751d126eddac8436338f8447b1967b5aa3b12a2b9a6e712e921039080ddc448e7fe4985676a7d2d6567dfc2febb8207a6f29b9737233f4c3b44d62103fa2d2c07653afb0a73340ee9e6b10ff0d4624c93b01127e9a69efbed48468e0121029669b79d7e05f20ca7aa418edc839da904467253c9e47194cb83e69ee418315155aeffffffff014a1a3101000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac00000000

Transaction fingerprint (md5):
07bda44b81863fec03e9a9a49f555a93
QR code for transaction written to transaction.png
